bee-common/src/shutdown.rs

```diff
@@ -26,12 +28,12 @@ pub enum Error {
 
 pub type ShutdownNotifier = oneshot::Sender<()>;
 pub type ShutdownListener = oneshot::Receiver<()>;
-pub type WorkerHandle = task::JoinHandle<Result<(), WorkerError>>;
+pub type WorkerHandle = dyn Future<Output = Result<(), WorkerError>> + Unpin;
```

Comment:
let's not call it handle anymore, but just `Worker`. Analogously (I couldn't speak out that word the first time), we now should have a type alias `Action` to make it consistent. So I propose:
```rust
pub type Worker = Box<dyn Future<Output = Result<(). WorkerError>> + Unpin>;
pub type Action = Box<dyn Fn()>;
```

Reply:
Sure